src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyWithoutShouldStartLine.java

```diff
@@ -0,0 +1,69 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyWithoutShouldStartLine {
+
+    private int a;
+    private static int b;
+    {  a = 2; }  //violation if option=ALONE
```

Comment:
@ps-sp , test verification become too complicated.
Please copy this file 2 times and use each of 3 files for each mode separately,  and define in line "// violation", withtout any "if".
In javadoc of input class , please define for what config it is used. 
In this it will be easy to validate work by simply review of in Input file (config defined in javadoc, expected violations inlined as tralling comments)

Reply:
Done. There are no tests for option `same` though.